engine/script/src/http_service.cpp

```diff
@@ -436,26 +429,42 @@ namespace dmHttpService
         dmMessage::URL url;
         url.m_Socket = http_service->m_Socket;
         dmMessage::Post(0, &url, 0, 0, (uintptr_t) dmHttpDDF::StopHttp::m_DDFDescriptor, 0, 0, 0);
+
+        // Stop the balancer first, so we don't accept any new requests
+        dmThread::Join(http_service->m_Balancer);
+
+
+        // Cancel them all first, as opposed to one-by-one
         for (uint32_t i = 0; i < http_service->m_Workers.Size(); ++i)
         {
             dmHttpService::Worker* worker = http_service->m_Workers[i];
+
             url.m_Socket = worker->m_Socket;
-            dmDNS::StopChannel(http_service->m_Workers[i]->m_DNSChannel);
             dmMessage::Post(0, &url, 0, 0, (uintptr_t) dmHttpDDF::StopHttp::m_DDFDescriptor, 0, 0, 0);
-// On iOS we detach() the thread on creation so that we don't have to wait for
-// it during shutdown (see above for reason why)
-#if !defined(DM_PLATFORM_IOS)
-            dmThread::Join(worker->m_Thread);
-#endif
+
+            worker->m_Canceled = 1;
```

Comment:
A pointer to this flag is given to each new dmHttp::Client(), and allows it to cancel before a timeout occurs.